### PR TITLE
Fix ESLint extension config

### PR DESCRIPTION
**Pull request by andrewbranch**
/cc @IllusionMH

Thanks @uniqueiniquity for the fix!

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,8 +6,11 @@
         }
     ],
     "eslint.options": {
-        "rulePaths": ["../scripts/eslint/built/rules/"],
+        "rulePaths": ["./scripts/eslint/built/rules/"],
```
Comment (uniqueiniquity) [inline .vscode/settings.json]: I'm surprised changing this allowed it to continue to work. For me, if the only change I make is the below, everything works fine.

Comment (andrewbranch) [reply .vscode/settings.json]: All the other paths are relative to the workspace root, though, so it kind of makes sense?

Comment (uniqueiniquity) [reply .vscode/settings.json]: When I do this locally, it breaks my setup :( it all depends on where the working directory of the eslint process is. Apparently on my machine, it's in the `src` folder, and it's not clear to me why. Maybe we should just set these to true and keep the leading `..` so it's unambiguous?

Comment (uniqueiniquity) [reply .vscode/settings.json]: Sorry, "set these to true" means `changeProcessCWD: true` in both cases.

Comment (IllusionMH) [reply .vscode/settings.json]: Same here. If I copy updated file - no error when I open `tests/cases/fourslash/addMemberInDeclarationFile.ts`, but fails in `src/services/findAllReferences.ts`
```
Error: ENOENT: no such file or directory, scandir 'c:\web\projects\TypeScript\src\scripts\eslint\built\rules'
```

If I change only `"eslint.workingDirectories"` it's reversed with error when `tests/cases/fourslash/addMemberInDeclarationFile.ts` is opened.

File: .vscode/settings.json [hunk 2]
```diff
@@ -6,8 +6,11 @@
...
         "ext": [".ts"]
     },
-    "eslint.workingDirectories": ["./src", "./scripts"]
+    "eslint.workingDirectories": [
```
Comment (IllusionMH) [inline .vscode/settings.json]: Is this options required for this repo?
I tried to remove it, used `"rulePaths": ["./scripts/eslint/built/rules/"]` and everything looks fine at first glance (for user who don't know what rules should be applied where):
 - `space-in-parens`, `@typescript-eslint/semi`, and custom `boolean-trivia` work in both `src` and `scripts` folders
 - `no-var` suppressed in `src\lib\esnext.bigint.d.ts`, but will be reported as error if I comment out supression.
 - no `ENOENT` errors when I open files in `tests` folder

Comment (uniqueiniquity) [reply .vscode/settings.json]: You're right - since we only have one ESLint config file, we shouldn't need this.
Works well on my box as you suggest.

Comment (IllusionMH) [reply .vscode/settings.json]: To be fair - there is second `src/.eslintrc.json` config (and third in same folder), but I tried to tweak it (commented out `no-var` suppression) and saw expected results. Looks like `files` in overrides were resolved as patterns relative to config anyway.

Comment (andrewbranch) [reply .vscode/settings.json]: Cool, that works for me too.

**Pull request merged**